src/applications/mhv-medical-records/util/pdfHelpers/blueButton.js

```diff
@@ -26,210 +26,222 @@ import { generateDemographicsContent } from './demographics';
+    conditions,
+    vitals,
+    medications,
+    appointments,
+    demographics,
+    militaryService,
+    accountSummary,
+  },
+  recordFilter,
+) => {
   const data = [];
 
-  if (isArrayAndHasItems(labsAndTests) && labsAndTests.length > 0) {
-    data.push({
-      type: recordType.LABS_AND_TESTS,
-      title: 'Lab and test results',
-      subtitles: [
-        'Most lab and test results are available 36 hours after the lab confirms them. Pathology results may take 14 days or longer to confirm.',
-        'If you have questions about these results, send a secure message to your care team.',
-        'Note: If you have questions about more than 1 test ordered by the same care team, send 1 message with all of your questions.',
-        `Showing ${labsAndTests.length} records from newest to oldest`,
-      ],
-      records: labsAndTests.map(record => {
-        const title = record.name;
-        let content;
-
-        if (record.type === labTypes.CHEM_HEM) {
-          content = generateChemHemContent(record);
-        }
-        if (record.type === labTypes.MICROBIOLOGY) {
-          content = generateMicrobioContent(record);
-        }
-        if (record.type === labTypes.PATHOLOGY) {
-          content = generatePathologyContent(record);
-        }
-        if (record.type === labTypes.EKG) {
-          content = generateEkgContent(record);
-        }
-        if (record.type === labTypes.RADIOLOGY) {
-          content = generateRadiologyContent(record);
-        }
-        return { title, ...content };
-      }),
-    });
-  }
-
-  if (notes?.length) {
-    data.push({
-      type: recordType.CARE_SUMMARIES_AND_NOTES,
-      title: 'Care summaries and notes',
-      subtitles: [
-        'This report only includes care summaries and notes from 2013 and later.',
-        'For after-visit summaries, (summaries of your appointments with VA providers), go to your appointment records.',
-      ],
-      records: notes.map(record => {
-        const title = record.name;
-        let content;
-
-        if (record.type === loincCodes.DISCHARGE_SUMMARY) {
-          content = generateDischargeSummaryContent(record);
-        }
-        if (
-          record.type === loincCodes.PHYSICIAN_PROCEDURE_NOTE ||
-          record.type === loincCodes.CONSULT_RESULT
-        ) {
-          content = generateProgressNoteContent(record);
-        }
-        return { title, ...content };
-      }),
-    });
-  }
-
-  if (vaccines?.length) {
-    data.push({
-      type: recordType.VACCINES,
-      title: 'Vaccines',
-      subtitles: [
-        'This list includes vaccines you got at VA health facilities and from providers or pharmacies in our community care network. It may not include vaccines you got outside our network.',
-        'For complete records of your allergies and reactions to vaccines, review your allergy records in this report.',
-      ],
-      records: generateVaccinesContent(vaccines),
-    });
-  }
-
-  if (allergies?.length) {
```

Comment:
Will this change leave the sections in place, even if there are no records?

Reply:
@mmoyer-va yep, because the PDF needs to know which sections were selected to be included by the user.